Dockerfile

```diff
@@ -0,0 +1,34 @@
+
+FROM node:8.16 as builder
+
+# Create build directory
+WORKDIR /tmp/build
+
+# Install first dependencies, so docker can cache this layer
+COPY package*.json truffle-config.js ./
```

Comment:
Unless we call install after this line there is no point in having multiple copies.